src/router.ts

```diff
@@ -41,15 +42,19 @@ const router = createRouter({
           name: "redirect",
           beforeEnter(to, from, next) {
             const store = useStore()
-            store
-              .fetch(Simulations.getLatestId())
-              .then(() => {
-                next(`/simulation${to.query.to || ""}`)
-              })
-              .catch(() => {
-                next("/")
-              })
+            const simulationLatestId = Simulations.getLatestId()
+            if (simulationLatestId) {
+              store
+                .fetch(simulationLatestId)
+                .then(() => {
+                  next(`/simulation${to.query.to || ""}`)
+                })
+                .catch(() => {
+                  next("/")
+                })
+            }
           },
+          redirect: "/",
```

Comment:
```suggestion
            }
            else {
              next("/")
          },
```
plutôt, non ? Ça permet d'avoir toute la logique regroupée.

Reply:
En fait je souhaitais corriger le warning qui requiert la définition de la propriété `redirect` dans l'interface `RouteRecordRedirect`. Je ne saisi pas pourquoi ce type là précisément était attendu pour cette route mais je viens de trouver plus simple en forçant la route à être de type `RouteRecordRaw` (qui n'a pas d'obligation sur la propriété `redirect`).

J'ajoute ta suggestion en plus car ce cas est potentiellement atteignable 👍